testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1834,6 +1844,69 @@ public void testCreateTable()
         assertFalse(getQueryRunner().tableExists(getSession(), tableNameLike));
     }
 
+    @DataProvider
+    public Object[][] likeTableParentTableCandidates()
+            throws Exception
+    {
+        return likeTableParentTableCandidatesData();
+    }
+
+    protected Object[][] likeTableParentTableCandidatesData()
+            throws Exception
+    {
+        String tableName = "test_create_orig_" + randomTableSuffix();
+        return new Object[][] {
+                {tableName, "CREATE TABLE " + tableName + " (a bigint, b double, c varchar(50))"}
+        };
+    }
+
+    @Test(dataProvider = "likeTableParentTableCandidates")
+    public void testCreateTableLikeIncludingProperties(String tableName, String createTableSql)
+    {
+        String catalogName = getSession().getCatalog().orElseThrow();
+        assertUpdate(createTableSql);
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+        assertTableColumnNames(tableName, "a", "b", "c");
+
+        String tableNameLike = "test_create_like_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableNameLike + " (LIKE " + tableName + " INCLUDING PROPERTIES, d bigint, e varchar(50))");
+        assertTrue(getQueryRunner().tableExists(getSession(), tableNameLike));
+        assertTableColumnNames(tableNameLike, "a", "b", "c", "d", "e");
+
+        String parentTableProperties = getTablePropertiesString(tableName);
+        String newTableProperties = getTablePropertiesString(tableNameLike);
+
+        Set<String> ignoredPropertiesInCreateLikeIncludingProperties = this.getQueryRunner().getTablePropertyManager().getAllProperties(new CatalogName(catalogName)).stream()
+                .filter(PropertyMetadata::isIgnoredInCreateTableLikeIncludingProperties)
+                .map(PropertyMetadata::getName)
+                .collect(toImmutableSet());
+
+        for (String property : ignoredPropertiesInCreateLikeIncludingProperties) {
+            Pattern p = Pattern.compile(property + " = (.*),?");
+            Matcher parentTableMatcher = p.matcher(parentTableProperties);
+            Matcher newTableMatcher = p.matcher(newTableProperties);
+
+            if (newTableMatcher.find() && parentTableMatcher.find()) {
+                // When we ignore the property, it should not be the same in the new table
+                assertNotSame(newTableMatcher.group(0), parentTableMatcher.group(0));
```

Comment:
This may not be necessarily true.
If both the parent and child tables have default values for a table property, their values for that property will be the same.

Reply:
I see your point, however there is no real case right now. It could be fixed by comparing it to the default value within the metadata for that property.